tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -31,6 +32,7 @@ internal class TraceContext
         public TraceContext(IDatadogTracer tracer, TraceTagCollection tags = null)
         {
             Tracer = tracer;
+            IastRequestContext = Iast.Iast.Instance.Settings.Enabled ? new IastRequestContext() : null;
```

Comment:
When IAST is enabled, this will allocate several object (`IastRequestContext`, `VulnerabilityBatch`, `List<Vulnerability>`) for every trace even if vulnerabilities are never found. Can we be more lazy and delay this instantiation until it's needed?

Reply:
I have improved that. Now, the iastContext is only instantiated when adding the first vulnerability.

Reply:
Thanks!